File: Leetcode/C++/48. Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& A) {
        int n = A.size();
        
        for(int i=0; i<n; i++) {
            auto l = 0, r = i;
            while(l < r) {
                swap(A[r][l], A[l][r]);
                l++; r--;
            }
            
            l = i;
            r = n-1;
            while(i && l<r) {
                swap(A[r][l], A[l][r]);
                l++; r--;
            }
        }
        
        for(auto &v : A) reverse(v.begin(), v.end());
    }
};

File: Leetcode/Rust/48. Rotate Image.rs
impl Solution {
    pub fn rotate(A: &mut Vec<Vec<i32>>) {
        A.reverse();
        
        for i in 0..A.len() {
            for j in 0..i {
                let t = A[i][j];
                A[i][j] = A[j][i];
                A[j][i] = t;
            }
        }
    }
}
